Added SecurityConfig class

Petro-Smoliar opened the pull request:
Implemented UserDetails interface
Added the implementation for UserDetailsService interface Added Role entity
Added Liquibase changesets

Petro-Smoliar committed:
Added SecurityConfig class
Implemented UserDetails interface
Added the implementation for UserDetailsService interface
Added Role entity
Added Liquibase changesets

okuzan commented on src/main/java/com/example/book/store/config/SecurityConfig.java:
```diff
@@ -0,0 +1,41 @@
+                   .userDetailsService(userDetailsService)
+                   .build();
+    }
+
```
```suggestion
```

okuzan commented on src/main/java/com/example/book/store/config/SecurityConfig.java:
```diff
@@ -0,0 +1,41 @@
+                   .csrf(AbstractHttpConfigurer::disable)
+                   .authorizeHttpRequests(
+                       auth -> auth
+                                   .requestMatchers("/api/auth/registration")
```
.requestMatchers("/auth/**", "/swagger-ui/**", "/v3/api-docs/**")

we shouldn't filter auth/login too

okuzan commented on src/main/resources/db/changelog/db.changelog-master.yaml:
```diff
@@ -3,4 +3,11 @@ databaseChangeLog:
+  - include:
+      file: db/changelog/change/05-add-roles.yaml
+  - include:
+      file: db/changelog/change/06-add-user_admin.yaml
```
![image](https://github.com/Petro-Smoliar/book_store/assets/49767049/6a3346ea-4b45-4cae-ba1c-226b5901e015)

Petro-Smoliar committed:
Added SecurityConfig class
Implemented UserDetails interface
Added the implementation for UserDetailsService interface
Added Role entity
Added Liquibase changesets
